core/src/account_api.rs

```diff
@@ -77,4 +191,30 @@ mod integration_tests {
             }
         }
     }
+
+    #[test]
```

Comment:
Ways to improve test:
+ Make a `Clock` which gives `AuthService` time, and mock it out here
+ Hold the `key` constant and then assert specific values
+ Make sure your tests cover all the branches of your code